scubainit/src/entfiles.rs

```diff
@@ -0,0 +1,70 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader, Write};
+use std::marker::PhantomData;
+
+use crate::util::short_write;
+
+pub trait Entry: Sized {
+    fn from_line(line: &str) -> Option<Self>;
+    fn to_line(&self) -> String;
+}
+
+pub struct EntFileReader<'a, T> {
+    reader: BufReader<&'a File>,
+    marker: PhantomData<T>, // T must be used
+}
```

Comment:
> I generally try to stay away from using lifetimes as much as possible as they're generally a lot more difficult to reason about than ownership.

I agree. This was done out of (perceived) necessity. The reason is because I wanted to re-use the same open file for both the `EntFileReader` and `EntFileWriter` (which operate sequentially on that file). I couldn't find a way to do this without making `BufReader` use a File *reference*, which required introducing a lifetime.

Reply:
I tried this but ran into an issue: see #239.

Reply:
Merged #239 into this branch.